src/Symfony/Component/DependencyInjection/Definition.php

```diff
@@ -171,6 +171,10 @@ public function getDecoratedService()
      */
     public function setClass($class)
     {
+        if ($class instanceof Parameter) {
```

Comment:
if we add the deprecation, it should not only check for Parameter but anything that is not string. Currently the class accepts anything as a class. A Parameter instance was used in a broken test but that is purely random and a user could use any class that has a `__toString` method.

Reply:
Yes, I could add another check for everything that is not string or null.